### PR TITLE
Update operator-sdk to v0.8.0

### PR DESCRIPTION
**Pull request by kevinrizza**
As part of normal maintenance, update operator-sdk reference to latest
version

Replication steps:
* Installed latest v0.8.0 operator-sdk binary onto local environment
* Ran `operator-sdk print-deps  --as-file` and copied output
* Replaced top portion of gopkg.toml with `print-deps` output
* Ran `dep ensure -update` from root of operator-marketplace repository to
install dependencies

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -41,11 +58,16 @@ required = [
...
+
 [[constraint]]
   name = "github.com/operator-framework/operator-sdk"
   # The version rule is used for a specific release and the master branch for in between releases.
   # branch = "master" #osdk_branch_annotation
-  version = "=v0.3.0" #osdk_version_annotation
+  version = "=v0.8.0" #osdk_version_annotation
 
 # Everything above this is required by the SDK. For each SDK version upgrade, the
 # above contents should be replaced by code copied from the SDK repo located at:
```
Comment (awgreene) [inline Gopkg.toml]: This block should be updated to reflect using the operator-sdk binary to generate the dependencies.

Comment (kevinrizza) [reply Gopkg.toml]: Ah, this is what you were talking about. Fixed in latest merge.

File: Gopkg.toml [hunk 2]
```diff
@@ -1,38 +1,55 @@
...
+  name = "k8s.io/kube-openapi"
+  revision = "0cf8f7e6ed1d2e3d47d02e3b6e559369af24d803"
+
+[[override]]
+  name = "github.com/go-openapi/spec"
+  branch = "master"
+
+[[override]]
+  name = "sigs.k8s.io/controller-tools"
+  revision = "9d55346c2bde73fb3326ac22eac2e5210a730207"
```
Comment (tkashem) [inline Gopkg.toml]: I assume these are from indirect references needed by the sdk?

Comment (kevinrizza) [reply Gopkg.toml]: Yeah, they're indirect references. I ran a command (shown slightly lower in this file, as well as in the commit message) from the sdk binary that gives you all of these references.

File: Gopkg.toml [hunk 3]
```diff
@@ -41,15 +58,20 @@ required = [
...
   name = "github.com/operator-framework/operator-sdk"
   # The version rule is used for a specific release and the master branch for in between releases.
   # branch = "master" #osdk_branch_annotation
-  version = "=v0.3.0" #osdk_version_annotation
+  version = "=v0.8.0" #osdk_version_annotation
 
 # Everything above this is required by the SDK. For each SDK version upgrade, the
-# above contents should be replaced by code copied from the SDK repo located at:
-# https://github.com/operator-framework/operator-sdk/blob/8f0fe6975128d3670e92e0509c645d80657a6d68/Gopkg.toml#L9-L23
+# above contents should be replaced by running the command `operator-sdk print-deps --as-file`
```
Comment (awgreene) [inline Gopkg.toml]: +1!

**Pull request merged**